OptimizelySDK.Tests/OptimizelyTest.cs

```diff
@@ -240,6 +240,94 @@ public void TestCreateUserContextMultipleAttribute()
+                { "flagKey", featureKey },
+                { "enabled", true },
+                { "variables", variables.ToDictionary() },
+                { "variationKey", variation.Key },
+                { "ruleKey", ruleKey },
+                { "reasons", reasons },
+                { "decisionEventDispatched", false }
+            }))), Times.Once);
+        }
+
+        [Test]
+        public void TestDecisionNotificationSentWhenSendFlagDecisionsTrueAndRollout()
+        {
+            var featureKey = "boolean_single_variable_feature";
+            var featureFlag = Config.GetFeatureFlagFromKey(featureKey);
+            var variables = Optimizely.GetAllFeatureVariables(featureKey, TestUserId);
+            var reasons = new OptimizelyDecideOption[0];
+            var userAttributes = new UserAttributes
+            {
+                { "device_type", "iPhone" },
+                { "location", "San Francisco" }
+            };
+            var experiment = Config.GetRolloutFromId("166660").Experiments[1];
+            var ruleKey = experiment.Key;
+            var variation = Config.GetVariationFromKey(experiment.Key, "177773");
+            Config.SendFlagDecisions = true;
+            var fallbackConfigManager = new FallbackProjectConfigManager(Config);
+            var optimizely = new Optimizely(fallbackConfigManager,
+                NotificationCenter,
+                EventDispatcherMock.Object,
+                LoggerMock.Object,
+                ErrorHandlerMock.Object,
+                null,
+                new ForwardingEventProcessor(EventDispatcherMock.Object, NotificationCenter, LoggerMock.Object, ErrorHandlerMock.Object),
+                null);
+
+            // Mocking objects.
+            NotificationCallbackMock.Setup(nc => nc.TestDecisionCallback(It.IsAny<string>(), It.IsAny<string>(),
+               It.IsAny<UserAttributes>(), It.IsAny<Dictionary<string, object>>()));
+            optimizely.NotificationCenter.AddNotification(NotificationCenter.NotificationType.Decision, NotificationCallbackMock.Object.TestDecisionCallback);
+
+            var optimizelyUserContext = optimizely.CreateUserContext(TestUserId, userAttributes);
+            optimizelyUserContext.Decide(featureKey);
+            NotificationCallbackMock.Verify(nc => nc.TestDecisionCallback(DecisionNotificationTypes.FLAG, TestUserId, userAttributes, It.Is<Dictionary<string, object>>(info => TestData.CompareObjects(info, new Dictionary<string, object> {
+                { "flagKey", featureKey },
+                { "enabled", true },
+                { "variables", variables.ToDictionary() },
+                { "variationKey", variation.Key },
+                { "ruleKey", ruleKey },
+                { "reasons", reasons },
+                { "decisionEventDispatched", true }
+            }))), Times.Once);
+        }
```

Comment:
Can you validation that event is actually sent out and event contents as well?

Reply:
Also, can we add the same couple of tests for "featureTest" (not rollout) as well?